exporters/otlp/otlpmetric/internal/metrictransform/metric.go

```diff
@@ -160,41 +157,21 @@ func transformer(ctx context.Context, exportSelector export.ExportKindSelector,
 // Any errors encoutered transforming input will be reported with an
 // ErrTransforming as well as the completed ResourceMetrics. It is up to the
 // caller to handle any incorrect data in these ResourceMetrics.
-func sink(ctx context.Context, in <-chan result) ([]*metricpb.ResourceMetrics, error) {
+func sink(ctx context.Context, res *resource.Resource, in <-chan result) ([]*metricpb.ResourceMetrics, error) {
```

Comment:
This change here is the only functional change in this PR, the optimization mentioned in the PR description is that this grouping function is now 1-level instead of 2-levels as discussed in https://github.com/open-telemetry/opentelemetry-go/issues/2119.